// @framer/blindmetal.new-component-page/code/LottieCustom.tsx
import * as React from "react"
import {
    Frame,
    addPropertyControls,
    ControlType,
    Data,
    Override,
    RenderTarget,
} from "framer"
import Lottie from "react-lottie"
import * as new_account_light from "./new_account_light.json"
import * as new_project_light from "./new_project_light.json"
import * as new_employee_light from "./new_employee_light.json"

// Open Preview: Command + P
// Learn more: https://framer.com/api

LottieCustom.defaultProps = {}

export function LottieCustom(props) {
    var lottie: any = React.createRef()

    const [selected, setSelected] = React.useState(false)
    const [direction, setDirection] = React.useState(1)

    const value = props.value || "a"
    const animationType = {
        a: new_account_light,
        b: new_project_light,
        c: new_employee_light,
    }

    function Selected() {
        setSelected(true)
        setDirection(1)
    }

    function UnSelected() {
        setSelected(false)
        setDirection(0)
    }

    return (
        <div onMouseEnter={Selected} onMouseLeave={UnSelected}>
            <Lottie
                ref={ref => (lottie = ref)}
                direction={direction ? 1 : -1}
                options={{
                    loop: false,
                    autoplay: selected,
                    animationData: animationType[value],
                }}
                height={"100%"}
                width={"100%"}
            />
        </div>
    )
}

addPropertyControls(LottieCustom, {
    value: {
        type: ControlType.Enum,
        defaultValue: "a",
        options: ["a", "b", "c"],
        optionTitles: ["New account", "New Project", "New employee"],
    },
})
